src/main/java/com/questhelper/quests/treegnomevillage/TreeGnomeVillage.java

```diff
@@ -0,0 +1,207 @@
+
+import com.questhelper.QuestDescriptor;
+import com.questhelper.QuestHelperQuest;
+import com.questhelper.Zone;
+import com.questhelper.panel.PanelDetails;
+import com.questhelper.questhelpers.BasicQuestHelper;
+import com.questhelper.requirements.ItemRequirement;
+import com.questhelper.steps.ConditionalStep;
+import com.questhelper.steps.NpcStep;
+import com.questhelper.steps.ObjectStep;
+import com.questhelper.steps.QuestStep;
+import com.questhelper.steps.conditional.*;
+import net.runelite.api.ItemID;
+import net.runelite.api.NpcID;
+import net.runelite.api.ObjectID;
+import net.runelite.api.coords.WorldPoint;
+
+import java.util.*;
+
+@QuestDescriptor(
+	quest = QuestHelperQuest.TREE_GNOME_VILLAGE
+)
+public class TreeGnomeVillage extends BasicQuestHelper
+{
+	private QuestStep talkToKingBolren, talkToCommanderMontai, bringWoodToCommanderMontai, talkToCommanderMontaiAgain,
+		firstTracker, secondTracker, thirdTracker, fireBallista, climbTheLadder, talkToKingBolrenFirstOrb,
+		talkToTheWarlord, fightTheWarlord, returnOrbs;
+
+	private ConditionForStep completeFirstTracker, completeSecondTracker, completeThirdTracker,
+		notCompleteFirstTracker, notCompleteSecondTracker, notCompleteThirdTracker;
+
+	private Conditions talkToFirstTracker, talkToSecondTracker, talkToThirdTracker, shouldFireBallista;
+
+	private ConditionalStep retrieveOrb;
+
+	Zone upstairsTower;
+	ZoneCondition isUpstairsTower;
+	ItemRequirement logRequirement;
+
+	private final int TRACKER_1_VARBITID = 599;
+	private final int TRACKER_2_VARBITID = 600;
+	private final int TRACKER_3_VARBITID = 601;
+
+	@Override
+	public Map<Integer, QuestStep> loadSteps()
+	{
+		setupSteps();
+		setupZones();
+		setupConditions();
+		setupItemRequirements();
+
+		return CreateSteps();
+	}
+
+	private Map<Integer, QuestStep> CreateSteps()
+	{
+		Map<Integer, QuestStep> steps = new HashMap<>();
+		steps.put(0, talkToKingBolren);
+		steps.put(1, talkToCommanderMontai);
+		steps.put(2, bringWoodToCommanderMontai);
+		steps.put(3, talkToCommanderMontaiAgain);
+		steps.put(4, talkToTrackersStep());
+		steps.put(5, retrieveOrbStep());
+		steps.put(6, talkToKingBolrenFirstOrb);
+		steps.put(7, defeatWarlordStep());
+		steps.put(8, returnOrbs);
+		return steps;
+	}
+
+	private QuestStep talkToTrackersStep()
+	{
+		ConditionalStep talkToTrackers = new ConditionalStep(this, firstTracker);
+		talkToTrackers.addStep(talkToSecondTracker, secondTracker);
+		talkToTrackers.addStep(talkToThirdTracker, thirdTracker);
+		talkToTrackers.addStep(shouldFireBallista, fireBallista);
+		return talkToTrackers;
+	}
+
+	private QuestStep retrieveOrbStep()
+	{
+		retrieveOrb = new ConditionalStep(this, climbTheLadder, "Retrieve the first orb from chest.");
+		ObjectStep getOrbFromChest = new ObjectStep(this, 2183, new WorldPoint(2506, 3259, 1), "Retrieve the first orb from chest.");
+		retrieveOrb.addStep(isUpstairsTower, getOrbFromChest);
+		retrieveOrb.addSubSteps(getOrbFromChest, climbTheLadder);
```

Comment:
Having an additional step prior to climbing the ladder for climbing over the crumbling wall would be great, it can be a bit hard for a newer player to spot how to enter (likely they'd try entering via the door).

Reply:
It's kind of awkward doing a zone inside the building since the shape, it would need like 2-3 zones to be done properly? I will add a note in the step text to clarify to go through the crumbled wall